akka-persistence/src/main/scala/akka/persistence/Persistence.scala

```diff
@@ -146,6 +147,16 @@ class Persistence(val system: ExtendedActorSystem) extends Extension {
 
   private val config = system.settings.config.getConfig("akka.persistence")
 
+  /**
+   * INTERNAL API: When starting many persistent actors at the same time the journal
+   * its data store is protected from being overloaded by limiting number
+   * of recoveries that can be in progress at the same time.
+   */
+  @InternalApi private[akka] val recoveryPermitter: ActorRef = {
+    val maxPermits = config.getInt("max-concurrent-recoveries")
+    system.systemActorOf(RecoveryPermitter.props(maxPermits), "recoveryPermitter")
+  }
```

Comment:
I thought about that also, but concluded that it wasn't worth the additional complexity and sometimes you actually want to limit the total number recoveries as it is implemented now.